### ISSUE TITLE
Pointer to non-const object and pack facilities

### ISSUE THREAD

**Issue by damb** [EarthScope/libmseed#106]
Hi there,

I'm working on [Rust bindings for libmseed](https://github.com/damb/mseed). While implementing the pack facilities I noted that the data structs to be packed are non-const, e.g.

```c
int msr3_pack(MS3Record* msr, ...)
int msr3_repack_mseed3(MS3Record* msr, ...)
int msr3_pack_header3(MS3Record* msr, ...)
int msr3_pack_header2(MS3Record* msr, ...)
int64_t mstl3_pack(MS3TraceList* mstl, ...)
```

This is due to setting default values for both the `reclen` and the `encoding` attribute. However, from a user point of view it seems odd that the object to be packed is modified while being packed/serialized. I.e. in particular in languages like Rust where variables are *const* by default.

What about passing pointers to const objects and using default values if required without modifying the object?

**Comment by chad-earthscope** [comment]
> I'm working on [Rust bindings for libmseed](https://github.com/damb/mseed).

Ah, that explains the `const` focus.  I was planning to do that too, I'm glad you're on it 👍 

> What about passing pointers to const objects and using default values if required without modifying the object?

That's a good idea, I do not see any issues with that for the msr3 routines.

One exception is `mstl3_pack()`, which modifies the contained buffers by default (it can be turned off with the `MST_MAINTAINMSTL` flag).  This supports the pattern of using the `MS3TraceList` as a "rolling buffer" for continuous creation of records in a streaming fashion.

**Comment by damb** [comment]
> That's a good idea, I do not see any issues with that for the msr3 routines.

:+1: Do you want me to provide a PR?

**Comment by damb** [comment]
> > I'm working on [Rust bindings for libmseed](https://github.com/damb/mseed).
> 
> Ah, that explains the `const` focus. I was planning to do that too, I'm glad you're on it :+1:

OT: Of course, feedback is welcome.

**Comment by chad-earthscope** [comment]
I think this is done in c05b4e4301ec462d9081632f4f733ec19c24bc8f.

I would appreciate a review before rolling another release after which the API is going to be frozen.

**Comment by damb** [comment]
> I think this is done in [c05b4e4](https://github.com/EarthScope/libmseed/commit/c05b4e4301ec462d9081632f4f733ec19c24bc8f).

Thanks a lot, @chad-earthscope.

The changes look good to me.

What about adding a test case to https://github.com/EarthScope/libmseed/blob/57092169e08f69a3da98555ee19c265bac00eea7/test/test-write.c that tests the default behaviour, i.e. both `msr-reclen` and `msr->encoding` are set to `-1`?

**Comment by chad-earthscope** [comment]
> What about adding a test case

Yup, that's what I was missing, thanks.  Added in 732da972b7494d3bece70e1e40bb17bfd76e9ff8
